README.md

```diff
@@ -45,7 +48,7 @@ The optional parameters are as follows:
 ### Reference
 
 - `--refgenome`: a URI to the reference genome to use during pipeline analysis
-- `--reference_sample_id`: the sample identifier of a sample in the samplesheet that contains a provided `reference_assembly` to use as a reference genome during pipeline analysis
+- `--reference_sample_id`: the sample identifier of a sample (`sample` or `sample_name`) in the samplesheet that contains a provided `reference_assembly` to use as a reference genome during pipeline analysis
```

Comment:
Curious: is there a case where the `reference_sample_id` could match 1 `sample` and 1 `sample_name`? Same IDs but different rows?

Reply:
This is the function that returns the `reference_assembly`
```JAVA
def select_reference(refgenome, reference_sample_id, sample_assemblies) {

    if(refgenome) {
        reference_genome = Channel.value(file(refgenome))
        log.debug "Selecting reference genome ${reference_genome} from '--refgenome'."
    }
    else if (reference_sample_id) {
        // Check each meta category (meta.id, meta.id_alt, meta.irida_id) for a match to params.reference_sample_id
        reference_genome = sample_assemblies.filter { (it[0].id == reference_sample_id || it[0].irida_id == reference_sample_id || it[0].id_alt == reference_sample_id) && it[1] != null}
                                            .ifEmpty { error("The provided reference sample ID (${reference_sample_id}) is either missing or has no associated reference assembly.") }
                                            .map { it[1] }
                                            .first()
        log.debug "Selecting reference genome ${reference_genome} from '--reference_sample_id'."
    }
    else {
        error("Unable to select a reference. Neither '--refgenome' nor '--reference_sample_id' were provided.")
    }

    return reference_genome
```
If two rows match `reference_sample_id` AND have a reference_assembly value, then the function simply returns the first. Someone could theoretically assign two different assemblies by accident. We could make an error for this. Otherwise it shouldn't be a problem.

Reply:
Thanks for bringing this up Eric, and thanks for describing the behaviour Steven.

I think this particular situation would be incredibly rare. It would require someone to 1) select at least two different reference genomes from the drop-down in IRIDA Next and 2) enter in an identifier that could potentially match both of them. Someone could still disambiguate this case by only selecting one of the assemblies from the drop-down in the input samplesheet in IRIDA Next.

So, I think what we have right now is a good solution.

Reply:
If I'm understanding the process correctly, would they be selecting two or more reference genomes or two or more samples to run through the pipeline (which may have reference assemblies associated with them automatically loaded in by Irida Next)? It seems like some Irida Next use-cases involve "loading everything up and running everything", and these collisions might occur once users are using lots of samples that have been assembled at some point.

Reply:
Thanks Eric. They would have to select two or more references. IRIDA Next does not auto-populate the reference assembly column since it's not marked as required. Hence, someone would have to go out of their way to setup a scenario where there are multiple matches for a sample identifier/sample name.

Reply:
Aah, okay, gotcha.